lib/rubocop/cop/style/access_modifier_declarations.rb

```diff
@@ -63,6 +63,7 @@ class AccessModifierDeclarations < Cop
 
         def on_send(node)
           return unless node.access_modifier?
+          return if node.parent.pair_type?
```

Comment:
Seems to me the real problem here is that the cop is inspecting all `send` node instead of just going over whatever is in a `class/module` body. Your fix will do, but I think the cop's design cop be improved in general.

Reply:
Sure! I think it ’s a good improvement idea.
`block` may also need to be considered in addition to `class` and `module` (#7298). Since redesign may take some time, I'd like to proceed with another PR. What do you think?

Reply:
> block may also need to be considered in addition to class and module

You're right.

> Since redesign may take some time, I'd like to proceed with another PR. What do you think?

Yeah, that's fine by me. We can merge this one in the mean time.